src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1125,98 +1134,123 @@ public override Stream EndGetRequestStream(IAsyncResult asyncResult)
-                    }
-                }
+                    // Create empty content so that we can send the entity-body header.
+                    _sendRequestMessage.Content ??= new ByteArrayContent(Array.Empty<byte>());
 
-                request.Headers.TransferEncodingChunked = SendChunked;
-
-                if (KeepAlive)
-                {
-                    request.Headers.Connection.Add(HttpKnownHeaderNames.KeepAlive);
+                    _sendRequestMessage.Content.Headers.TryAddWithoutValidation(headerName, _webHeaderCollection[headerName!]);
                 }
                 else
                 {
-                    request.Headers.ConnectionClose = true;
+                    _sendRequestMessage.Headers.TryAddWithoutValidation(headerName, _webHeaderCollection[headerName!]);
                 }
+            }
 
-                if (_servicePoint?.Expect100Continue == true)
-                {
-                    request.Headers.ExpectContinue = true;
-                }
+            if (_servicePoint?.Expect100Continue == true)
+            {
+                _sendRequestMessage.Headers.ExpectContinue = true;
+            }
+
+            _sendRequestMessage.Version = ProtocolVersion;
 
-                request.Version = ProtocolVersion;
+            _sendRequestMessage.Headers.TransferEncodingChunked = SendChunked;
+
+            if (KeepAlive)
+            {
+                _sendRequestMessage.Headers.Connection.Add(HttpKnownHeaderNames.KeepAlive);
+            }
+            else
+            {
+                _sendRequestMessage.Headers.ConnectionClose = true;
+            }
 
-                _sendRequestTask = async ?
-                    client.SendAsync(request, _allowReadStreamBuffering ? HttpCompletionOption.ResponseContentRead : HttpCompletionOption.ResponseHeadersRead, _sendRequestCts!.Token) :
-                    Task.FromResult(client.Send(request, _allowReadStreamBuffering ? HttpCompletionOption.ResponseContentRead : HttpCompletionOption.ResponseHeadersRead, _sendRequestCts!.Token));
+            _sendRequestMessage.Version = ProtocolVersion;
+            HttpCompletionOption completionOption = _allowReadStreamBuffering ?
+                            HttpCompletionOption.ResponseContentRead : HttpCompletionOption.ResponseHeadersRead;
+            _sendRequestTask = async ?
+                _httpClient.SendAsync(_sendRequestMessage, completionOption, _sendRequestCts!.Token) :
+                Task.Run(() => _httpClient.Send(_sendRequestMessage, completionOption, _sendRequestCts!.Token));
 
-                HttpResponseMessage responseMessage = await _sendRequestTask.ConfigureAwait(false);
+            return _sendRequestTask!;
+        }
 
-                HttpWebResponse response = new HttpWebResponse(responseMessage, _requestUri, _cookieContainer);
+        private async Task<WebResponse> HandleResponse(bool async)
+        {
+            _sendRequestTask ??= SendRequest(async);
```

Comment:
This should be `??=`, because basically, we're starting a request on `RequestStream` if we call it, but if we call `GetResponse` right away, we're starting the request here.